### ISSUE TITLE
Realtime GI or support for lightmap baker?

### ISSUE THREAD

**Issue by VinnyVicious** [AtomicGameEngine/AtomicGameEngine#1664]
Is there a way of improving lighting in my scenes?

**Comment by Alan-FGR** [comment]
Hello.
Unfortunately the light baking system we had in the works (glow) doesn't have a maintainer anymore, so unless you're OK coding some RTGI stuff (which got considerably easier now that Blender has probes and cubemap bakers you could take advantage of, and there are snippets in GitHub), then I'd say the best way to proceed is to just bake the lightmap using a 3D app of your choice (Blender is quite good for that). There's a sample shader that supports lightmaps, but in any case it's very simple to do once you have the baked data.

**Comment by nem0** [comment]
So can I export the scene from editor to Blender?

**Comment by Alan-FGR** [comment]
@nem0 that would only make sense if you were using the editor to make your scenes, like to create the level meshes and place the static stuff and such, since dynamic objects you don't want baked. But I'm assuming here that the scenes are made in a 3rd party application, especially considering the overall usability of the current editor for such purposes.

**Comment by nem0** [comment]
AFAIK Atomic does not support making levels in 3rd party modelers, you can only make geometry, but can't create gameobjects or attach components in it. So this flow is not really possible.

**Comment by Alan-FGR** [comment]
If you need to somehow script stuff in the nodes holding the static stuff in your levels, then nope that's not possible, there's however a blender exporter that writes the scene xml files so objects in blender are translated into nodes with a few extras, including a sample on how to use that for lightmapping. If you need to script areas in your world, you can just use extra nodes for that, you normally don't need to attach scripts to static level stuff, you can easily workaround that by just using other nodes. You can't of course place enemies and such in Blender, but you can make all the static stuff in your level.

**Comment by nem0** [comment]
For example: 
static geometry, a house
with dynamic geometry: doors

**Comment by Alan-FGR** [comment]
Yes, that's an inconvenience, not a dealbreaker: you have the house including walls and everything 'static' in Blender, then you place the doors and whatnot in Atomic. I honestly don't see the big problem in that, it's not a showstopper by any means. 😕

**Comment by nem0** [comment]
The you change the scene in blender, reexport and you lose your changes from editor?

**Comment by Alan-FGR** [comment]
Nope, obviously not, you just replace the node with data from Blender 😮

**Comment by nem0** [comment]
How? That's not possible at the moment AFAIK

**Comment by Alan-FGR** [comment]
🤦‍ keep the data from Blender in a separate XML, say 'house_static_stuff.xml' when you change say a wall in the house, you update that XML, say you moved a wall and rebaked the lighting data, now you gotta move the doors and objects and whatnot to match that. You just don't touch the data in 'house_static_stuff.xml' unless when exporting from Blender, and Blender cannot change the objects it don't know about, you're not replacing your scene with the data in 'house_static_stuff.xml', it's just some extra data in your scene that's coming from an external XML, think of it like a Node with a Mesh, when you change the Mesh the other nodes in the scene are still going to be there, but the Mesh will be updated, it's not going to remove all the nodes, that's not how it works 😕

**Comment by nem0** [comment]
Let's move this to chat.
